sub-packages/bionemo-esm2/src/bionemo/esm2/model/finetune/datamodule.py

```diff
@@ -15,23 +15,118 @@
 
 
 import functools
+import os
+from typing import Sequence, Tuple, Union
 
-import pytorch_lightning as pl
+import numpy as np
+import pandas as pd
 import torch
 import torch.utils.data
 from nemo.lightning.pytorch.plugins import MegatronDataSampler
 from nemo.utils import logging
 from pytorch_lightning.utilities.types import EVAL_DATALOADERS, TRAIN_DATALOADERS
+from torch import Tensor
+from torch.utils.data import Dataset
 
 from bionemo.core.data.multi_epoch_dataset import IdentityMultiEpochDatasetWrapper, MultiEpochDatasetResampler
 from bionemo.esm2.data import tokenizer
 from bionemo.esm2.model.finetune.finetune_regressor import InMemorySingleValueDataset
 from bionemo.esm2.model.finetune.finetune_token_classifier import InMemoryPerTokenValueDataset
 from bionemo.llm.data import collate
+from bionemo.llm.data.datamodule import MegatronDataModule
+from bionemo.llm.data.types import BertSample
 from bionemo.llm.utils.datamodule_utils import infer_num_samples
 
 
-class ESM2FineTuneDataModule(pl.LightningDataModule):
+class InMemoryCSVDataset(Dataset):
```

Comment:
I'd rather you separate the file IO from the dataset logic; especially if this is all just in-memory anyways.

Maybe just `ProteinDataset` or `ESMProteinDataset` that takes in `sequences: pd.Series, labels: pd.Series` or something like that. Then you could have a small convenience function outside the class (or a `@classmethod` alternative constructor) that creates this from a file path specifically.